src/graph/graph_op.cc

```diff
@@ -653,4 +654,77 @@ DGL_REGISTER_GLOBAL("graph_index._CAPI_DGLMapSubgraphNID")
     *rv = GraphOp::MapParentIdToSubgraphId(parent_vids, query);
   });
 
+#if !defined(WIN32) && !defined(_WIN32) && !defined(__WIN32__) && !defined(__NT__)
```

Comment:
Let's move all the new codes to a separate file `graph/metis_partition.cc`

Reply:
Is it necessary to guard so many macros? In [dmlc-core](https://github.com/dmlc/dmlc-core/blob/692216e8633eacae001a023ebbdcafbad0985854/include/dmlc/filesystem.h), we only guard _WIN32.